python/hail/tests/test_api.py

```diff
@@ -420,6 +420,15 @@ def test_joins(self):
         kt2 = kt2.annotate_globals(kt_foo=kt.index_globals().foo)
         self.assertEqual(kt2.globals.kt_foo.value, 5)
 
+    def test_interval_join(self):
+        left = hl.utils.range_table(50, n_partitions=10)
+        intervals = hl.utils.range_table(4)
+        intervals = intervals.key_by(interval = hl.interval(intervals.idx * 10, intervals.idx * 10 + 5))
+        left = left.annotate(interval_matches = intervals.index(left.key))
+        self.assertTrue(left.all(hl.case()
+                                 .when(left.idx % 10 < 5, left.interval_matches.idx == left.idx // 10)
+                                 .default(hl.is_missing(left.interval_matches))))
+
```

Comment:
I think we should also have a test using the bracket syntax.

Reply:
I want to push back on this -- we test the bracket syntax elsewhere, and it just calls `.index()`. It doesn't feel like we need to test bracket syntax everywhere we test `index`. Thoughts?

Reply:
As discussed out of line, __getitem__ is simple enough in the case that you're just passing an expression or list of expressions that this is OK with me.